webapp/channels/src/components/threading/virtualized_thread_viewer/virtualized_thread_viewer.tsx

```diff
@@ -350,6 +353,12 @@ class ThreadViewerVirtualized extends PureComponent<Props, State> {
         const isLastPost = itemId === this.props.lastPost.id;
         const isRootPost = itemId === this.props.selected.id;
 
+        const post = useSelector((state: GlobalState) => getPost(state, this.props.selected.id));
+        const getThreadOrSynthetic = useMemo(makeGetThreadOrSynthetic, [post.id]);
```

Comment:
There should be no need to re-do the selector on `post.id` changes, so we can drop the dependency.